hoot-core/src/main/cpp/hoot/core/algorithms/WaySplitter.cpp

```diff
@@ -80,7 +81,9 @@ vector<WayPtr> WaySplitter::createSplits(const vector<WayLocation>& wl)
   WayLocation end(_map, _a, _a->getNodeCount() - 1, 0.0);
   if (last.compareTo(end) != 0)
   {
-    result[result.size() - 1] = WaySubline(last, end).toWay(_map, _nf.get());
+    WayPtr w = WaySubline(last, end).toWay(_map, _nf.get());
+    w->setPid(_a->getId());
+    result[result.size() - 1] = w;
```

Comment:
Everything above this was a zero sum game, nothing changed with removing duplicated code.  These are the changes that affected the remaining test files.  When creating splits in the `WaySplitter` since the original isn't always deleted set the parent ID for all of the splits to the original way ID.  That way if the original way still exists, the `WayJoiner::joinParentChild` function will rejoin them.  If the original way is removed then the `WayJoiner::joinSiblings` function will rejoin them.